src/convert.cpp

```diff
@@ -112,3 +112,64 @@ ddwaf_object* to_ddwaf_object(ddwaf_object *object, Napi::Env env, Napi::Value v
   // we use empty maps for now instead of null. See issue !43
   return ddwaf_object_map(object);
 }
+
+Napi::Value from_ddwaf_object(ddwaf_object *object, Napi::Env env) {
+  DDWAF_OBJ_TYPE type = object->type;
+
+  Napi::Value result;
+
+  switch (type) {
+    case DDWAF_OBJ_SIGNED:
+      result = Napi::Number::New(env, object->intValue);
+      break;
+    case DDWAF_OBJ_UNSIGNED:
+      result = Napi::Number::New(env, object->uintValue);
+      break;
+    case DDWAF_OBJ_STRING:
+      result = Napi::String::New(env, object->stringValue, object->nbEntries);
+      break;
+    case DDWAF_OBJ_ARRAY: {
+      Napi::Array arr = Napi::Array::New(env, object->nbEntries);
+
+      if (env.IsExceptionPending()) {
+        mlog("Exception pending");
```

Comment:
I'd rather have a maximum amount of data decoded and silently fail with nulls for undecipherable data wdyt ?

Reply:
So, why would it fail here? I am afraid something has to go very wrong to get there.

Reply:
These errors should never happen when the WAF is okay.
I think throwing a JS exception for those cases so that the library knows something unexpected happened during the call is the way good to go.